src/scram.js

```diff
@@ -127,15 +127,24 @@ function generate_cnonce() {
 }
 
 /**
- * @typedef {Object} Password
- * @property {string} Password.name
- * @property {string} Password.ck
- * @property {string} Password.sk
- * @property {number} Password.iter
- * @property {string} salt
+ * @typedef {Object} SCRAMKey
+ * @property {string} SCRAMKey.name
+ * @property {string} SCRAMKey.ck
+ * @property {string} SCRAMKey.sk
+ * @property {number} SCRAMKeyiter
+ * @property {string} SCRAMKey.salt
  */
 
 const scram = {
+    test: function (connection, hashName, hashBits) {
+        return true; // XXX debug
+        return connection.authcid !== null
+            && (
+                (typeof connection.pass === 'string' || connection.pass instanceof String)
+                || (connection.pass?.name === hashName)
+            );
+    },
```

Comment:
This is here to make sure that we don't accidentally try to SCRAM-SHA-1 when we only have a HT-SHA-256-NONE token available.